app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -119,6 +122,27 @@ def ip_addresses
       []
     end
 
+    def data_disks
+      return nil unless @azure_vm.storage_profile.data_disks
+      disks = @azure_vm.storage_profile.data_disks
+    end
+
+    def volumes_attributes=(attrs)
+    end
+
+    def volumes
+      unless data_disks.nil?
+        volumes = []
+        data_disks.each do |disk|
```

Comment:
If we return an empty array in previous comment, we will not have to worry about `nil` in `data_disks`, so we can rewrite this method to return:

``` ruby
@volumes ||= data_disks.map do |disk|
  OpenStruct.new(:disk => disk, :persisted? => true)
end
```
This way we will return the same array over and over again